subsys/net/lib/aws_fota/src/aws_fota.c

```diff
@@ -248,40 +273,52 @@ static int get_job_execution(struct mqtt_client *const client,
 						      file_path, sizeof(file_path),
 						      &execution_version_number);
 
-	if (err < 0) {
-		LOG_ERR("Error when parsing the json: %d", err);
-		goto cleanup;
-	} else if (err == 0) {
+	if (err == AWS_FOTA_JSON_RES_SKIPPED) {
 		LOG_DBG("Got only one field");
 		LOG_DBG("No queued jobs for this device");
 		return 0;
+	} else if (err < 0
+		   && err != AWS_FOTA_JSON_RES_INVALID_DOCUMENT
+		   && err != AWS_FOTA_JSON_RES_URL_TOO_LONG) {
+		LOG_ERR("Error when parsing the json: %d", err);
+		err = -ENODATA;
+		goto cleanup;
 	}
 
 	/* Check if the incoming job is already being handled. */
 	if (strncmp(job_id_incoming, job_id_handling, sizeof(job_id_incoming)) == 0) {
 		LOG_WRN("Job already being handled, ignore message");
 		err = 0;
 		goto cleanup;
-	} else {
-		strncpy(job_id_handling, job_id_incoming, sizeof(job_id_handling));
-		job_id_handling[sizeof(job_id_handling) - 1] = '\0';
 	}
 
+	set_current_job_id(job_id_incoming);
+
+	/* Check if the update data is valid */
+	if (err == AWS_FOTA_JSON_RES_INVALID_DOCUMENT) {
+		LOG_ERR("Invalid FOTA update document: %d", err);
+		return set_current_job_failed(client);
+	} else if (err == AWS_FOTA_JSON_RES_URL_TOO_LONG) {
+		LOG_ERR("URL elements too long for buffer: %d", err);
+		return set_current_job_failed(client);
+	}
```

Comment:
No need to check for the other parse errors?

Reply:
That's being handled above (in the section of your previous comment), where we deal with all other errors except `AWS_FOTA_JSON_RES_INVALID_DOCUMENT` and `AWS_FOTA_JSON_RES_URL_TOO_LONG`.